File: DeepSpaceJava/src/deepspace/SpaceCity.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;


/**
 *
 * @author samuel
 */
public class SpaceCity extends SpaceStation {
    private SpaceStation base;
    private ArrayList<SpaceStation> collaborators;
    
    SpaceCity(SpaceStation base, ArrayList<SpaceStation> rest){
        super(base);
        collaborators = rest;
        this.base = base;
    }
    
    public ArrayList<SpaceStation> getCollaborators(){
        return collaborators;
    }
    
    public float fire(){
        float fire=0;
        fire += super.fire();
        for(int i=0; i<collaborators.size();++i){
            fire += collaborators.get(i).fire();
        }
        return fire;
    }
    
    public float protection(){
        float pro=0;
        pro += super.protection();
        for(int i=0; i<collaborators.size();++i){
            pro += collaborators.get(i).protection();
        }
        return pro;  
    }
    
    public Transformation setLoot(Loot loot){
           super.setLoot(loot);
           return Transformation.NOTRANSFORM;
    }
    
    public SpaceCityToUI getUIversion(){
        return new SpaceCityToUI(this);
    }
}

File: DeepSpaceRuby/lib/SpaceCity.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Deepspace
  require_relative'Loot'
  require_relative'Transformation'
  require_relative 'SpaceCityToUI'
  
class SpaceCity < SpaceStation
  attr_reader :collaborators
  
  #ss es tipo SpaceStation, rest array de spaceStation
  def initialize(ss,rest)
    super(ss.name,ss.ammoPower, ss.fuelUnits, ss.shieldPower, ss.weapons, ss.shieldBoosters,ss.hangar,ss.nMedals, ss.pendingDamage)
    @collaborators = Array.new
    for i in (0...rest.size)
      @collaborators.push(rest[i])
    end    
    @base=ss
  end
  
  def getUIversion
    SpaceCityToUI.new(self)
  end
  
  def fire
    fire=0.0
    fire+=super
    for i in(0...@collaborators.size)
      fire+=@collaborators[i].fire
    end
  end
  
  def protection
    pr=0.0
    pr+=super
    for i in(0...@collaborators.size)
      pr+=@collaborators[i].protection
    end
  end
  
  def setLoot(loot)
    super(loot)
    Transformation::NOTRANSFORM
  end
  
  
  
end
end